.github/workflows/cleanup.yml

```diff
@@ -1,14 +1,20 @@
 name: cleanup
 
 on:
-  schedule:
-    - cron: '*/5 * * * *'
+  # Triggerred on every PR close
+  pull_request:
+    types: [ "closed" ]
 
 jobs:
   cancel:
     runs-on: ubuntu-latest
+    timeout-minutes: 5
     steps:
-      - uses: n1hility/cancel-previous-runs@2e3c1893986568a2197c41957b9c809cbcf1a61e
+      - name: 'Cancel Runs For Closed PRs'
+        uses: styfle/cancel-workflow-action@0.9.0
         with:
-          token: ${{ github.token }}
-          workflow: ci.yml
+          # Cancel workflow when PR closed. https://github.com/styfle/cancel-workflow-action#advanced-ignore-sha
+          ignore_sha: true
+          # Note: workflow_id can be a Workflow ID (number) or Workflow File Name (string). Here we don't have ID.
```

Comment:
we are using ID below, hence this comment is not accurate. Why can't we use filename instead? Would be more readable